src/main/kotlin/nmcp/NmcpPublishTask.kt

```diff
@@ -62,23 +70,111 @@ abstract class NmcpPublishTask : DefaultTask() {
 
         val publicationType = publicationType.orElse("USER_MANAGED").get()
 
-        Request.Builder()
+        val endpoint = endpoint.getOrElse("https://central.sonatype.com/api/v1/")
+
+        val deploymentId = Request.Builder()
             .post(body)
             .addHeader("Authorization", "UserToken $token")
-            .url(endpoint.getOrElse("https://central.sonatype.com/api/v1/publisher/upload") + "?publishingType=$publicationType")
+            .url(endpoint + "publisher/upload?publishingType=$publicationType")
             .build()
             .let {
-                OkHttpClient.Builder()
-                    .addInterceptor(HttpLoggingInterceptor().apply {
-                        level = HttpLoggingInterceptor.Level.HEADERS
-                    })
-                    .build()
-                    .newCall(it).execute()
+                client.newCall(it).execute()
             }.use {
                 if (!it.isSuccessful) {
-                    error("Cannot publish to maven central (status='${it.code}'): ${it.body?.string()}")
+                    error("Cannot deploy to maven central (status='${it.code}'): ${it.body?.string()}")
                 }
+
+                it.body!!.string()
             }
+
+        logger.lifecycle("Nmcp: deployment bundle '$deploymentId' uploaded.")
+
+        if (verifyStatus.get()) {
+            logger.lifecycle("Nmcp: verifying deployment status...")
+            while (true) {
```

Comment:
It would be nice to configure a timeout with default being 5 or 10 minutes

Reply:
Added in https://github.com/GradleUp/nmcp/pull/27